doc/guides/optimizations/index.rst

```diff
@@ -0,0 +1,11 @@
+.. _optimizations:
+
+Optimizations
+##############
+
+Guides on how to optimize Zephyr for performance, power and footprint.
+
```

Comment:
We definitely need a section for MPU/MMU, here, as well. Disabling MPU improves runtime performance, aw well.